### PR TITLE
update glibc to 2.34

### LINKED ISSUE #1675: Update libglib to 2.69.0+
Before 2.69.0, libglib used a bundled copy of libpcre.  We had a dependency on libpcre2, but it [only supports libpcre (v1, i.e. 8.x)](https://gitlab.gnome.org/GNOME/glib/-/issues/1085) so it was building and using its own bundled version.

In 2.69.0, they removed the bundled libpcre and instead use a meson wrapdb build of libpcre.

To update, it seems like we'll need to either (1) package libpcre v1, or (2) make meson wrapdb builds work in buildsys, i.e. fetching its expected artifacts early or otherwise making it believe it doesn't need network access.

### LINKED ISSUE #1717: update to glibc 2.34
[glibc 2.34 is out!](https://lwn.net/Articles/864920/)

We'll also want the fix for [CVE-2021-38604](https://sourceware.org/git/?p=glibc.git;a=commit;h=b805aebd42364fe696e417808a700fdb9800c9e8).

### PR DESCRIPTION
**Pull request by bcressey**
**Issue number:**
Fixes #1717
Fixes #1675

**Description of changes:**
Update glibc to 2.34. Change the patching strategy to include patches added to the maintenance branch after the official 2.34 release, which aligns with Fedora's current approach.

Fix a cross-compilation bug for kubelet that caused us to link with target libraries for host programs when the architectures were the same. This was exposed by missing glibc symbols following the update, but was a latent regression from when we switched to the new upstream method of specifying the toolchain in #1612. It does not apply to older builds of kubelet, which used a local patch instead.

Add the upstream patch for chrony's seccomp filter so it runs with the newer glibc.

Update glib2 to 2.70.0, which includes a fix for the newer glibc. Tweak the cross-compiling file for meson to avoid attempts to execute target files, which failed because of missing glibc symbols.

**Testing done:**
Verified that `chronyd` starts running and stays up for multiple days on both architectures. Using the developer image, confirmed that `chronyc` can issue commands like "dump" without causing a seccomp violation.

Tested `aws-k8s-1.21` for both architectures and verified that `sonobuoy` passed. Confirmed that tasks could run on an `aws-ecs-1` build.

Tested `vmware-k8s-1.21` with a cluster created by EKS Anywhere. Verified that `sonobuoy` passed.

**Terms of contribution:**

By submitting this pull request, I agree that this contribution is dual-licensed under the terms of both the Apache License, version 2.0, and the MIT license.

### CODE REVIEW

**Review by etungsten — DISMISSED** [review]

**Comment by etungsten** [comment]
vmware builds are failing due to `libglib` package needed by `open-vm-tools`

File: packages/glibc/glibc.spec [hunk 1]
```diff
@@ -1,14 +1,47 @@
...
 Summary: The GNU libc libraries
 License: LGPL-2.1-or-later AND (LGPL-2.1-or-later WITH GCC-exception-2.0) AND GPL-2.0-or-later AND (GPL-2.0-or-later WITH GCC-exception-2.0) AND BSD-3-Clause AND ISC
 URL: http://www.gnu.org/software/glibc/
 Source0: https://ftp.gnu.org/gnu/glibc/glibc-%{version}.tar.xz
 Source1: glibc-tmpfiles.conf
-Patch1: glibc-cs-path.patch
-Patch2: glibc-c-utf8-locale.patch
-Patch1001: 1001-move-ldconfig-cache-to-ephemeral-storage.patch
+
+# Upstream patches from 2.34 release branch
```
Comment (tjkirch) [inline packages/glibc/glibc.spec]:
```suggestion
# Upstream patches from 2.34 release branch:
# ```
# git checkout origin/release/2.34/master
# git format-patch glibc-2.34..
# ```
```

**Comment by bcressey** [comment]
Back to draft to fix `open-vm-tools`.

**Review by tjkirch — COMMENTED** [review]
LGTM modulo the vmware build issue.

File: packages/libglib/libglib.spec [hunk 2]
```diff
@@ -1,10 +1,14 @@
...
 Release: 1%{?dist}
 Summary: The GLib libraries
-License: LGPL-2.1-or-later
+# glib2 is LGPL-2.1-only
+# pcre is BSD-3-Clause
+License: LGPL-2.1-only AND BSD-3-Clause
 URL: https://www.gtk.org/
-Source0: https://download.gnome.org/sources/glib/2.68/glib-%{version}.tar.xz
+Source0: https://download.gnome.org/sources/glib/2.70/glib-%{version}.tar.xz
+Source1: https://ftp.pcre.org/pub/pcre/pcre-8.37.tar.bz2
```
Comment (tjkirch) [inline packages/libglib/libglib.spec:11]:
```suggestion
# Note: the pcre version is specified in the glib archive in subprojects/libpcre.wrap
Source1: https://ftp.pcre.org/pub/pcre/pcre-8.37.tar.bz2
```

**Review by tjkirch — APPROVED** [review]
I left a couple suggestions to help future packagers keep these up to date, but overall it looks good to me.

**Review by arnaldo2792 — APPROVED** [review]
:motor_boat:

**Review by etungsten — APPROVED** [review]

**Review by zmrow — APPROVED** [review]
:kangaroo:

**Pull request merged**